bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/runtime/ShellExecutorHelper.java

```diff
@@ -39,29 +45,49 @@ public ShellExecutorHelper(Map<String, String> environment) {
 
     /**
      * Returns the full output.
+     * @param cmd Bash script, executed with bash -c "{script}".
+     * @param substitutionMap A <String, File> map for substituting the paths in the commands. use '${FILE}' for full
+     * compatibility.
+     * @return The script's output in an Optional wrapper.
      */
-    public Optional<String> quickReturnOutput(String script) {
-    	return this.quickReturnOutput(script, 1000);
+    public Optional<String> quickReturnOutput(String cmd, @Nullable Map<String, File> substitutionMap) {
```

Comment:
Maybe we could combine this quicReturnOutput into one method, with timeout and with an option to log to the console.
A renaming would be nice too, I do not see why it is quick ;)

Reply:
I have an idea for simplificaton of this section. A wrapper around the popular process exectuion libraries. With Gson like interface for easy usage, but with a factory for complex behaviour customization. Would be a separate project tho.